core/src/main/java/cucumber/runtime/formatter/PluginFactory.java

```diff
@@ -93,19 +93,29 @@ public Object create(String pluginString) {
     }
 
     private <T> T instantiate(String pluginString, Class<T> pluginClass, String pathOrUrl) throws IOException, URISyntaxException {
+        if (pathOrUrl == null) {
+            Constructor<T> constructor = findConstructor(pluginClass, null);
+            if (constructor != null) {
+                try {
+                    return constructor.newInstance();
+                } catch (InstantiationException e) {
+                    throw new CucumberException(e);
+                } catch (IllegalAccessException e) {
+                    throw new CucumberException(e);
+                } catch (InvocationTargetException e) {
+                    throw new CucumberException(e.getTargetException());
+                }
+            }
+        }
         for (Class ctorArgClass : CTOR_ARGS) {
             Constructor<T> constructor = findConstructor(pluginClass, ctorArgClass);
             if (constructor != null) {
                 Object ctorArg = convertOrNull(pathOrUrl, ctorArgClass, pluginString);
                 try {
-                    if (ctorArgClass == null) {
```

Comment:
If you add a `&& ctorArg == null` here you don't have to treat the null case separately.